configs/containers/docker-ubuntu-clang-mpich.yaml

```diff
@@ -0,0 +1,192 @@
+    eckit@1.19.0, ecmwf-atlas@0.29.0 ~trans ~fftw, eigen@3.4.0,
+    fckit@0.9.5, flex@2.6.4, fms@release-jcsda, g2@3.4.5, g2tmpl@1.10.0,
+    gftl-shared@1.5.0, hdf5@1.12.1, hdf@4.2.15, ip@3.3.3, jasper@2.0.32, jedi-cmake@1.3.0,
+    libpng@1.6.37, nccmp@1.9.0.1, netcdf-c@4.8.1, netcdf-cxx4@4.3.1,
+    netcdf-fortran@4.5.4, nlohmann-json-schema-validator@2.1.0, nlohmann-json@3.10.5,
+    parallel-netcdf@1.12.2, parallelio@2.5.4, py-numpy@1.22.3, py-pandas@1.4.0, py-scipy@1.8.0,
+    py-shapely@1.8.0, sp@2.3.3, udunits@2.2.28, w3nco@2.4.1, nco@5.0.6,
+    yafyaml@0.5.1, zlib@1.2.12, odc@1.4.5, crtm@v2.3-jedi.4]
+    # Don't build ESMF and MAPL for now:
+    # https://github.com/JCSDA-internal/MPAS-Model/issues/38
+    # https://github.com/NOAA-EMC/spack-stack/issues/326
+    # esmf@8.3.0b09, mapl@2.12.3
+
+  container:
+
+    # Select the format of the recipe e.g. docker,
+    # singularity or anything else that is currently supported
+    format: docker
+    # How to use:
+    #$ spack containerize > Dockerfile
+    #$ docker build -t myimage .
+    #$ docker run -it myimage
+
+    #format: singularity
+    # How to use:
+    #$ spack containerize > hdf5.def
+    #$ sudo singularity build hdf5.sif hdf5.def
+
+    # Sets the base images for the stages where Spack builds the
+    # software or where the software gets installed after being built..
+    images:
+      os: ubuntu:20.04
+      spack:
+        url: https://github.com/noaa-emc/spack
+        ref: jcsda_emc_spack_stack
+        resolve_sha: false
+
+    # Whether or not to strip binaries
+    strip: false
+
+    ## Additional system packages that are needed at runtime
+    os_packages:
+      build:
+      - clang-10
+      - libclang-10-dev
+      - libc++-10-dev
+      - cpp
+      - g++
+      - gcc
+      - gfortran
+      - git
+      - git-lfs
+      - make
+      - qt5-default
+      - libqt5svg5-dev
+      - qt5dxcb-plugin
+      - wget
+
+      final:
+      - clang-10
+      - libclang-10-dev
+      - libc++-10-dev
+      - cpp
+      - g++
+      - gcc
+      - gfortran
+      - git
+      - git-lfs
+      - make
+      - qt5-default
+      - libqt5svg5-dev
+      - qt5dxcb-plugin
+      - wget
+      # Implicitly included in build step
+      - build-essential
+      - ca-certificates
+      - curl
+      - file
+      - gnupg2
+      - iproute2
+      - locales
+      - python3
+      - python3-pip
+      - python3-setuptools
+      - unzip
+      - vim
+
+    # Extra instructions
+    extra_instructions:
+      pre_build: |
+        #Create symbolic links for clang compilers
+        RUN cd /usr/bin && \
+        ln -svf clang-10 clang && \
+        ln -svf clang++-10 clang++ && \
+        ln -svf clang-cpp-10 clang-cpp && \
+        cd /usr/lib/llvm-10/lib && \
+        ln -svf libc++abi.so.1.0 libc++abi.so
+      pre_final: |
+        #Set environment variables for installing tzdata
+        ENV DEBIAN_FRONTEND=noninteractive && \
+        TZ=Etc/UTC && \
```

Comment:
This like doesn't need `&& \`. However, after removing `&& \` I still get and error with docker build command complaining about extra `&&` in the Dockerfile. I had remove the space between `&&` and `\` in for it to work, which doesn't make much sense. So with this I didn't get the error about extra `&&`
```
ENV DEBIAN_FRONTEND=noninteractive &&\
TZ=Etc/UTC
```

Reply:
ok, I figured it out. You don't need `&&` in ENV. So it should be
```
        ENV DEBIAN_FRONTEND=noninteractive 
        TZ=Etc/UTC
```